core/modules/flowgen.cc

```diff
@@ -14,8 +14,12 @@
 
 typedef std::pair<uint64_t, struct flow *> Event;
 typedef std::priority_queue<Event, std::vector<Event>,
-                            std::function<bool(Event, Event)>>
-    EventQueue;
+                            std::function<bool(Event, Event)>> EventQueue;
+
+const Commands FlowGen::cmds = {
+  {"update", "FlowGenArg", MODULE_CMD_FUNC(&FlowGen::CommandUpdate), 1}
```

Comment:
While the command specifies .mt_safe==1, this CommandUpdate function is not quite thread-safe. Worker threads may see changes in the middle of packet processing, although the current code seems unlikely to crash.

I understand that "stop-the-world" is not desirable especially for packet generators. Some modules (e.g., Source) use ACCESS_ONCE to achieve atomic update. Since this module needs to update multiple variables at once, you will need more sophisticated mechanisms. For example, you can group all control variables as a struct, and atomically update a pointer to the struct with RCU.